crates/ty_python_semantic/src/types/generics.rs

```diff
@@ -671,6 +673,23 @@ impl<'db> SpecializationBuilder<'db> {
                 }
             }
 
+            (
+                Type::NominalInstance(NominalInstanceType {
+                    class: ClassType::Generic(formal_alias),
+                }),
+                Type::NominalInstance(NominalInstanceType {
+                    class: ClassType::Generic(actual_alias),
+                }),
+            ) if formal_alias.origin(self.db) == actual_alias.origin(self.db) => {
```

Comment:
what about generic protocol instances? Do we need to add a branch for them as well?

Reply:
Good idea, though I'd like to tackle that as a follow-on PR